src/ng/animate.js

```diff
@@ -369,6 +369,19 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * );
        * ```
        *
+       * <div class="alert alert-warning">
+       * **Note**: Generally, the events that are fired correspond 1:1 to `$animate` method names,
+       * e.g. {@link ng.$animate#addClass addClass} will fire `addClass`, and {@link ng.ngClass}
+       * will fire `addClass` if classes are added and `removeClass` if a classes are removed.
+       * However, there are two exceptions:
+       * <ul>
+       *   <li>if both an `addClass` and a `removeClass` action are performed during the same animation,
+       *   the event fired will be `setClass`. This is true even for `ngClass`.</li>
+       *   <li>an {@link ng.$animate#animate animate} call that only adds or removes classes will fire
+       * ` animate`, but if classes are added and removed at the same time, it will fire `setClass`.</li>
```

Comment:
Typo: _\` animate_ --> _ \`animate_ (no whitespace after \`)
Also, I am not sure what this means. Will it fire `setClass` instead of `animate` or will it fire both?

Reply:
It will fire only setClass. Will clarify.